invenio_rdm_records/resources/resources.py

```diff
@@ -677,7 +678,7 @@ def create_url_rules(self):
 
     def _get_record_with_files(self):
         uuid = resource_requestctx.view_args["uuid"]
-        return self.service.read_record(uuid=uuid, identity=g.identity)
+        return self.service.read_record(uuid=uuid, identity=g.identity).to_dict()
```

Comment:
Comment: Perhaps it's better to move this to the service method instead?

Reply:
but we always return a ResultItem from a service and we convert `to_dict()` in the resource layer no?